server/engine/telemetry.go

```diff
@@ -63,6 +63,8 @@ func PublishToSegment(db *gorm.DB) {
 	model.SetMetric(db, model.ApplicationId, config.GetEnvironment().APPLICATION_ID)
 	// time.RFC3339 format is the Go equivalent to ISO 8601 format (minus the milliseconds)
 	model.SetMetric(db, model.EndTime, time.Now().Format(time.RFC3339))
+	// time.RFC3339 format is the Go equivalent to ISO 8601 format (minus the milliseconds)
+	model.SetMetric(db, model.StartTime, time.Now().Format(time.RFC3339))
```

Comment:
StartTime should be set at the very beginning of the loop (and also checked and not set if already set).  It is intended to mark the start time of the deployment driver execution..  As for EndTime I am agnostic about putting it in this function or just before the call to it.  If you keep it here though, might be good to make it the very first thing in the function, followed by the other sets and then the publishing section.

Reply:
@bhavenst , we've changed the way we're retrieving the StartTime, we added this into `azure-rm` repo and now retrieving from there. Please let me know if it's ok now.

Reply:
Yeah that is fine, it could have gone either way.  We already have the marketplace start data so could have used that through the end time from deployment driver to get an overall time, but this is also fine.  Actually you might add one as I suggested as well with a name like deploymentdriverstarttime or something, then we can see the delay between them etc.  But up to you.